pkcs5/src/pbes2/kdf.rs

```diff
@@ -156,13 +166,13 @@ pub struct Pbkdf2Params<'a> {
 
 impl<'a> Pbkdf2Params<'a> {
     /// Initialize PBKDF2-SHA256 with the given iteration count and salt
-    pub fn hmac_with_sha256(iteration_count: u16, salt: &'a [u8]) -> Result<Self, CryptoError> {
```

Comment:
It would probably be good to leave this fallible.

I'm not sure why `u16` was used for the iteration count but it unnecessarily constrains the number of iterations. Using 100,000+ should be supported.

However, if this type is changed to a `u32`, it would also be good to set a sanity limit on the number of iterations.

RFC8018 says the sanity limit is implementation defined:

https://datatracker.ietf.org/doc/html/rfc8018#appendix-A.2

> iterationCount specifies the iteration count.  The maximum iteration count allowed depends on the implementation.  It is expected that implementation profiles may further constrain the bounds.